assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -93,25 +108,24 @@ function MediaProvider({ children }) {
   } = useConfig();
 
   const resetWithFetch = useCallback(() => {
+    // eslint-disable-next-line no-shadow
+    const { mediaType, pagingNum, searchTerm } = stateRef.current;
+
     resetFilters();
     if (!mediaType && !searchTerm && pagingNum === 1) {
       fetchMedia({ mediaType }, fetchMediaSuccess);
     }
-  }, [
-    fetchMedia,
-    fetchMediaSuccess,
-    mediaType,
-    pagingNum,
-    resetFilters,
-    searchTerm,
-  ]);
+  }, [fetchMedia, fetchMediaSuccess, resetFilters]);
 
   useEffect(() => {
     fetchMedia({ pagingNum, mediaType }, fetchMediaSuccess);
   }, [fetchMedia, fetchMediaSuccess, mediaType, pagingNum, searchTerm]);
 
   const uploadVideoPoster = useCallback(
     (id, src) => {
+      // eslint-disable-next-line no-shadow
+      const { processed, processing } = stateRef.current;
```

Comment:
This is, probably, not only good for performance, but actually the only valid way. Otherwise this method becomes racy. If someone keeps holding on to an old `uploadVideoPoster`, it also checks against the old `processed/processing`. But I have a different question here: what's the value of `processed` and `processing` being in the reducer state? These vars are both naturally look like `refs` to me. No one should be re-rendered because any of these values have changed. I.e. these do not at all belong in a rerenderable state of any kind: reducer or useState.

Reply:
This seems to work nicely. I've made this change here and rebased this PR to it: https://github.com/google/web-stories-wp/pull/2071